src/SpecialCase/ComposerJsonZendFrameworkPackageSpecialCase.php

```diff
@@ -0,0 +1,670 @@
+<?php
+
+/**
+ * @see       https://github.com/laminas/laminas-migration for the canonical source repository
+ * @copyright https://github.com/laminas/laminas-migration/blob/master/COPYRIGHT.md
+ * @license   https://github.com/laminas/laminas-migration/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Laminas\Migration\SpecialCase;
+
+use Composer\Semver\Semver;
+use RuntimeException;
+
+class ComposerJsonZendFrameworkPackageSpecialCase implements SpecialCaseInterface
+{
+    public function matches($filename, $content)
+    {
+        if (! preg_match('#\bcomposer\.json$#', $filename)) {
+            return false;
+        }
+
+        $composer = json_decode($content, true);
+        return isset($composer['require']['zendframework/zendframework'])
+            || isset($composer['require-dev']['zendframework/zendframework']);
```

Comment:
very minor: mixed case where ignored in composer v1, so technically we still can have `ZendFramework/ZendFramework` and it will be the same. But I see that changing it in the whole implementation is a bit pain, unless we "normalise" it first.

also, I cannot see any tests with `require-dev`.